### PR TITLE
Forward all claims in userinfo response

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Return all claims from OIDC user info server response as custom metadata headers.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
_NA_

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/3225

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/server/service.go [hunk 1]
```diff
@@ -200,6 +197,9 @@ func newHTTPServer(ctx context.Context, cfg *config.ServerConfig, _ *authConfig.
 		gwmuxOptions = append(gwmuxOptions, runtime.WithMetadata(auth.GetHTTPMetadataTaggingHandler()))
 	}
 
+	// This option sets subject in the user info response
+	gwmuxOptions = append(gwmuxOptions, runtime.WithForwardResponseOption(auth.GetUserInfoForwardResponseHandler()))
+
```
Comment (EngHabu) [inline pkg/server/service.go:202]: Doesn't need to move anymore, right?

File: auth/identity_context.go [hunk 2]
```diff
@@ -48,6 +51,14 @@ func (c IdentityContext) UserInfo() *service.UserInfoResponse {
...
 	}
 
+	userInfo := c.userInfo
+	if c.claims != nil {
+		m, err := structpb.NewStruct(*c.claims)
+		if err != nil {
+			logger.Errorf(context.TODO(), "Failed to marshal claims [%+v] to struct: %v", c.claims, err)
+		}
+		userInfo.AdditionalClaims = m
+	}
```
Comment (EngHabu) [inline auth/identity_context.go:61]: Can we move this to the constructor NewIdentityContext? no need to do this every time...

Comment (EngHabu) [reply auth/identity_context.go:61]: Also, let's refactor the constructor to return an Error and don't silence the error from here?
Also, I think we can remove the error log as long as you augment the error you will return with the information

File: pkg/server/service.go (hunk 1 above)
Comment (katrogan) [reply pkg/server/service.go]: done, thanks

**Review by EngHabu — APPROVED** [review]

**Pull request merged**